### ISSUE TITLE
Semi-colons toggle on/off when saving with `ts/member-delimiter-style`

### ISSUE THREAD

**Issue by jclaessens97** [antfu/eslint-config#393]
### Describe the bug

When saving on format semi-colons rule in typescript declarations toggle on/off. It also switches. 

For example when you have

```
interface Foo {
     bar: string
}

interface Bar {
   foo: string;
}
```
and `cmd + s`, it will add a `;` no the first `bar` property, and removes it on the `foo` property of the 2nd interface.

https://github.com/antfu/eslint-config/assets/10177148/edba7279-081b-4553-84d8-5d29bf5ef969

I used this config intensively on a previous project (before the flat config) and it worked beautifully, so I'm pretty sure it's not intended, but also might be a configuration thing on my side? 🤔 

### Reproduction

https://github.com/jclaessens97/eslint-semi

### System Info

```Shell
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 97.23 MB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.asdf/installs/nodejs/20.10.0/bin/node
    npm: 10.2.4 - ~/.asdf/plugins/nodejs/shims/npm
    pnpm: 8.14.1 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 120.0.6099.216
    Safari: 17.2.1
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
Can you try removing the two custom rules, instead, use this?

https://github.com/antfu/eslint-config#customization

```ts
antfu({
	stylistic: {
	  semi: true
	}
})
```

**Comment by jclaessens97** [comment]
That still removes the `semi` at the end of the interface members, but does add `semi` on other statements.

**Comment by jclaessens97** [comment]
The following config does it for me. Thanks. 

```typescript
antfu({
    stylistic: {
      semi: true,
      overrides: {
        'style/member-delimiter-style': [
          'error',
          {
            multiline: {
              delimiter: 'semi',
              requireLast: true,
            },
            singleline: {
              delimiter: 'semi',
              requireLast: true,
            },
            multilineDetection: 'brackets',
          },
        ],
      },
    },
})
```

**Comment by antfu** [comment]
This should be fixed with v2.6.3 with https://github.com/eslint-stylistic/eslint-stylistic/pull/267
